pkg/kubelet/kubeletconfig/util/files/files_test.go

```diff
@@ -0,0 +1,293 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package files
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+
+	utiltest "k8s.io/kubernetes/pkg/kubelet/kubeletconfig/util/test"
+	utilfs "k8s.io/kubernetes/pkg/util/filesystem"
+)
+
+const (
+	prefix = "test-util-files"
+)
+
+type file struct {
```

Comment:
@tianshapjq this file is what I'm thinking for the files tests, let me know what you think

Reply:
TBH, why do you try to test the Create(), Write() and MkdirAll() of the DefaultFs again? it's already been done in the files.go, and we just have to write some unit tests to the functions in files.go IMH.
BTW, you may need to remove the files created by this test case.

Reply:
> TBH, why do you try to test the Create(), Write() and MkdirAll() of the DefaultFs again? it's already been done in the files.go, and we just have to write some unit tests to the functions in files.go IMH.

I'm not sure I follow? This file is designed to unit test the functions in files.go.

> BTW, you may need to remove the files created by this test case.

Agree, will add some cleanup.

Reply:
sorry I didn't make it clear. My point is, you have tested the Create(), Write() and MkdirAll() here, seems my test case #57220 is useless, because my test case just invokes these functions as well. So, is my test case still needed?

Reply:
I understood #57220 as an effort to test the other utility functions, not the filesystem functions (`FileExists`, `EnsureFile`, `WriteTmpFile`, `ReplaceFile`, `DirExists`, `EnsureDir`).

Reply:
ok, I will update that pr after this merged~